python-frontend/src/main/java/org/sonar/python/cfg/ControlFlowGraphBuilder.java

```diff
@@ -186,13 +183,28 @@ private PythonCfgBlock build(Statement statement, PythonCfgBlock currentBlock) {
         return buildBreakStatement((BreakStatement) statement, currentBlock);
       case MATCH_STMT:
         return buildMatchStatement((MatchStatement) statement, currentBlock);
+      case FUNCDEF:
+        return buildFuncDefStatement((FunctionDef) statement, currentBlock);
       default:
         currentBlock.addElement(statement);
     }
 
     return currentBlock;
   }
 
+  private PythonCfgBlock buildClassDefStatement(ClassDef classDef, PythonCfgBlock successor) {
+    PythonCfgBlock block = build(classDef.body().statements(), successor);
+    block.addElement(classDef.name()); // represents binding to class name
+    classDef.decorators().stream().forEach(decorator -> successor.addElement(decorator.expression()));
```

Comment:
I have the feeling that if we wanted to be correct, we probably should add the `Decorator` tree as a distinct element (rather than just its expression), since the fact that it's part of a decorator has influence on the control flow (the underlying decorator function will be called here), which will not be represented by just having the `Name`.

However I see this approximation is already done for the class/function definitions themselves (although here the impact is more limited since it really is the creation of a symbol without other unexpected side effect).

Depending on how tricky this is to change without negative impact, I think we can either make the change directly, create a ticket to make the change later or just accept this approximation.